src/client/app/containers/view.mode.ts

```diff
@@ -27,43 +29,54 @@ import { Strings } from '../strings';
             this.snippet = snippet;
         });
     }
 
+    get host() {
+        return environment.current.host.toLowerCase();
+    }
+
     ngOnInit() {
         this.paramsSub = this._route.params
-            .subscribe(async(params) => {
-                if (environment.current.host.toUpperCase() !== params.host.toUpperCase()) {
-                    environment.current.host = params.host.toUpperCase();
+            .map(params => ({ type: params.type, host: params.host, id: params.id }))
+            .mergeMap(({ type, host, id }) => {
+                if (environment.current.host.toUpperCase() !== host.toUpperCase()) {
+                    environment.current.host = host.toUpperCase();
                     // Update environment in cache
                     environment.current = environment.current;
-                    await applyTheme(environment.current.host);
                 }
 
-                let urlSegments = this._route.snapshot.url;
-                switch (urlSegments[1].path) {
+                switch (type) {
                     case 'private-samples':
-                        let rawPrivateSamplesUrl = `${environment.current.config.samplesUrl}/private-samples/${params.host}/${params.segment}/${params.name}.yaml`;
-                        this._store.dispatch(new Snippet.ImportAction(Snippet.ImportType.SAMPLE, rawPrivateSamplesUrl, true));
-                        break;
-                    case 'gist':
-                        this._store.dispatch(new Snippet.ImportAction(Snippet.ImportType.GIST, params.id, true));
-                        break;
                     case 'samples':
-                        let rawSamplesUrl = `${environment.current.config.samplesUrl}/samples/${params.host}/${params.segment}/${params.name}.yaml`;
-                        this._store.dispatch(new Snippet.ImportAction(Snippet.ImportType.SAMPLE, rawSamplesUrl, true));
-                        break;
+                        let hostJsonFile = `${environment.current.config.samplesUrl}/view/${environment.current.host.toLowerCase()}.json`;
+                        return (this._request.get<JSON>(hostJsonFile, ResponseTypes.JSON)
+                            .map(data => ({ data: data, id: id }))
+                        );
+                    case 'gist':
+                        return Observable.of({ data: null, id: id});
                     default:
-                        break;
+                        return Observable.of({ data: null, id: null});
+                }
+            })
+            .subscribe(({ data, id }) => {
+                if (data && data[id]) {
+                    this._store.dispatch(new Snippet.ImportAction(Snippet.ImportType.SAMPLE, data[id], true));
+                } else if (id) {
+                    this._store.dispatch(new Snippet.ImportAction(Snippet.ImportType.GIST, id, true));
+                } else {
+                    // Redirect to error page
+                    window.localStorage.clear();
```

Comment:
What is this with clearing local storage?!  Sounds quite terrible!!!  Won't that lose folks' snippets, if they were, say, on Office Online?